service/gcs/runtime/runc/runc.go

```diff
@@ -376,7 +378,11 @@ func (r *runcRuntime) waitOnProcess(pid int) (oslayer.ProcessExitState, error) {
 }
 
 func (p *process) Wait() (oslayer.ProcessExitState, error) {
-	return p.c.r.waitOnProcess(p.pid)
+	state, err := p.c.r.waitOnProcess(p.pid)
+	if p.relay != nil {
```

Comment:
Ah, I missed this -- it already does for its init process but not the other processes. We could update this to keep track of the other processes and wait for them appropriately... But I think it actually doesn't really matter since we don't really care about the relays all finishing before we call back to HCS. We should keep an eye on this -- I'll add a note.